### ISSUE TITLE
cli: Program deploys are unreliable for large programs

### ISSUE THREAD

**Issue by jstarry** [solana-labs/solana#13829]
#### Problem
Program deploys with `solana deploy ..` fail for large programs (~400k bytes). This appears to be caused by sending transactions over TPU / RPC too quickly so that a large number of writes get dropped and need to be retried. Also, too many transaction status requests are sent too quickly which overload the RPC server and eventually time out.

#### Proposed Solution
- Throttle transaction sending
- Throttle transaction status requests

cc @CriesofCarrots

**Comment by jstarry** [comment]
To get the deploy working, I added a 20ms sleep in between each transaction send and a 200ms sleep between each status request. I came up with 20ms because with 1600 transactions, that added up to about 32s of total sleep time which was short enough to avoid expired blockhashes but in practice only about 3/4 of the transactions got confirmed (better than about 1/3 without the throttling change). I came up with 200ms by trial and error.
